### PR TITLE
Add named globbing using "**:name" or "{name:**}"

### PR DESCRIPTION
**Pull request by DerDackel**
Enhances the routing path matcher by allowing the binding of globs as a named parameter using the following syntax:

```java
get("/foo/**:tail", req -> {
  doSomething(req.params("tail")); // GET/foo/bar/baz -> tail == "bar/baz"
});

get("/foo/{tail:**}", req -> {
  doSomething(eq.params("tail")); // same
});

get("/foo/**:inbetween/end", req -> {
  doSomething(eq.params("inbetween")); // GET/foo/bar/baz/end -> inbetween == "bar/baz"
});

get("/foo/{inbetween:**}/end", req -> {
  doSomething(eq.params("inbetween")); // same
});
```

### CODE REVIEW

**Comment by jknack** [comment]
damn! you are fast!!

File: jooby/src/test/java/org/jooby/internal/RoutePatternTest.java [hunk 1]
```diff
@@ -139,6 +139,72 @@ public void any2() {
...
+        .matches("GET/com/a/foo/bar/moo/baz", (vars) -> {
+          assertEquals("a/foo", vars.get("first"));
+          assertEquals("moo/baz", vars.get("second"));
+        })
+        .butNot("GET/com/foo/baz")
+        .butNot("GET/test.jsp");
+  }
+
+  @Test
+  public void anyNamedMultiSpring() {
```
Comment (rodrigotechera) [inline jooby/src/test/java/org/jooby/internal/RoutePatternTest.java]: Is it worth testing any name multi with mixed syntax? Something like `new RoutePathAssert("GET", "com/**:first/bar/{second:**}")`

File: jooby/src/test/java/org/jooby/internal/RoutePatternTest.java [hunk 2]
```diff
@@ -139,6 +139,72 @@ public void any2() {
   }
 
   @Test
+  public void anyNamed() {
+    new RoutePathAssert("GET", "com/**:rest")
+        .matches("GET/com/test.jsp", vars -> assertEquals("test.jsp", vars.get("rest")))
+        .matches("GET/com/a/test.jsp", vars -> assertEquals("a/test.jsp", vars.get("rest")))
+        .matches("GET/com/", vars -> assertEquals("", vars.get("rest")))
+        .butNot("GET/test.jsp");
```
Comment (rodrigotechera) [inline jooby/src/test/java/org/jooby/internal/RoutePatternTest.java]: Is it worth testing `matches("GET/com", vars -> assertEquals("", vars.get("rest")))`? 
Notice that it doesn't has the last `/` in one of your test, it seems to me another corner case but not 100% sure.

Comment (DerDackel) [inline jooby/src/test/java/org/jooby/internal/RoutePatternTest.java]: I added that case deliberately, wanted to make sure the regex wasn't accidently including leading slashes on an empty match. Can change the behavior, but I find it to be more consistent, as that "/" is basically the leading slash of an empty path.

File: jooby/src/test/java/org/jooby/internal/RoutePatternTest.java (hunk 1 above)
Comment (DerDackel) [inline jooby/src/test/java/org/jooby/internal/RoutePatternTest.java]: Can't hurt! Added!

File: jooby/src/test/java/org/jooby/internal/RoutePatternTest.java [hunk 3]
```diff
@@ -139,6 +139,72 @@ public void any2() {
...
+  }
+
+  @Test
+  public void anyNamedInner() {
+    new RoutePathAssert("GET", "com/**:rest/bar")
+        .matches("GET/com/foo/bar", (vars) -> assertEquals("foo", vars.get("rest")))
+        .matches("GET/com/a/foo/bar", (vars) -> assertEquals("a/foo", vars.get("rest")))
+        .butNot("GET/com/foo/baz")
+        .butNot("GET/test.jsp");
```
Comment (rodrigotechera) [inline jooby/src/test/java/org/jooby/internal/RoutePatternTest.java]: Is it worth testing `matches("GET/com/foo/bars/bar", (vars) -> assertEquals("foo/bars", vars.get("rest")))`? Or `matches("GET/com/foo/abar/bar", (vars) -> assertEquals("foo/bars", vars.get("rest")))`
Again, it seems to me another corner case but not 100% sure.

File: jooby/src/test/java/org/jooby/internal/RoutePatternTest.java (hunk 2 above)
Comment (rodrigotechera) [inline jooby/src/test/java/org/jooby/internal/RoutePatternTest.java]: Hey, not saying you should remove any of your test. They seems to be good, just suggesting/asking for more test if they worth adding them.

Comment (DerDackel) [inline jooby/src/test/java/org/jooby/internal/RoutePatternTest.java]: I was just explaining the reasoning for that test case, while misreading a char in your example myself :) Will take care of that one too!
